plugins/inputs/slurm/slurm.go

```diff
@@ -0,0 +1,508 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package slurm
+
+import (
+	"context"
+	stdTls "crypto/tls"
+	_ "embed"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+
+	goslurm "github.com/pcolladosoto/goslurm/v0038"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Slurm struct {
+	URL              string          `toml:"url"`
+	Username         string          `toml:"username"`
+	Token            string          `toml:"token"`
+	IgnoredEndpoints []string        `toml:"ignored_endpoints"`
+	ResponseTimeout  config.Duration `toml:"response_timeout"`
+	tls.ClientConfig
+
+	client      *goslurm.APIClient
+	baseURL     *url.URL
+	endpointMap map[string]bool
+}
+
+func (*Slurm) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *Slurm) Init() error {
+	if s.ResponseTimeout == 0 {
+		s.ResponseTimeout = config.Duration(time.Second * 5)
+	}
```

Comment:
Would it make sense to allow for `s.ResponseTimeout = 0` to indicate an infinite timeout?

Reply:
As seen [here](https://pkg.go.dev/net/http#Client), a timeout of `0` would indeed configure an infinite timeout. Now, I'm nowhere near knowledgeable enough to decide whether that's going to work with Telegraf's 'event loop' or if it's going to cause some problems later down the road.

It's true the requests can also be cancelled explicitly through the context passed to the different API methods, but the current implementation leverages `context.Background` wrapped with the authentication values which means the context will never be cancelled.

Now, if you'd rather let users configure an infinite timeout I'd be more than happy to tweak the above lines: I'm just worried about getting in Telegram's way...

Reply:
Let me help you out here... Please set the `ResponseTimeout: config.Duration(5 * time.Second)` in `init()` (all lowercase) as this allows the user to overwrite the value in the config and set it to zero.

Reply:
Thanks a ton for the information! I added these changes on d4770953500843837cfec55385da2e7a8193c015.

Thanks again for the tip!